src/extensions/default/InAppNotifications/main.js

```diff
@@ -0,0 +1,372 @@
+                    dataType: "json",
+                    cache: false
+                }).done(function (notificationInfo, textStatus, jqXHR) {
+                    lastInfoURLFetchTime = (new Date()).getTime();
+                    PreferencesManager.setViewState("lastNotificationURLFetchTime", lastInfoURLFetchTime);
+                    PreferencesManager.setViewState("notificationInfo", notificationInfo);
+                    result.resolve(notificationInfo);
+                }).fail(function (jqXHR, status, error) {
+                    // When loading data for unit tests, the error handler is
+                    // called but the responseText is valid. Try to use it here,
+                    // but *don't* save the results in prefs.
+
+                    if (!jqXHR.responseText) {
+                        // Text is NULL or empty string, reject().
+                        result.reject();
+                        return;
+                    }
+
+                    try {
+                        data = JSON.parse(jqXHR.responseText);
+                        result.resolve(data);
+                    } catch (e) {
+                        result.reject();
+                    }
+                });
+            });
+        } else {
+            result.resolve(data);
+        }
+
+        return result.promise();
+    }
+
+
+    /**
+     * Check for notifications, notification overlays are always displayed
+     *
+     * @return {$.Promise} jQuery Promise object that is resolved or rejected after the notification check is complete.
+     */
+    function checkForNotification() {
+        var result = new $.Deferred();
+        var versionInfoUrl;
+
+        _getNotificationInformation(versionInfoUrl)
+            .done(function (notificationInfo) {
+                // Get all available notifications
+                var notifications = notificationInfo.notifications;
+                if (notifications && notifications.length > 0) {
+                    // Iterate through notifications and act only on the most recent
+                    // applicable notification
+                    notifications.every(function(notificationObj) {
+                        // Only show the notification overlay if the user hasn't been
+                        // alerted of this notification
+                        if (_checkNotificationValidity(notificationObj)) {
+                            if (notificationObj.silent) {
+                                // silent notifications, to gather user validity based on filters
+                                HealthLogger.sendAnalyticsData("notification", notificationObj.sequence, "handled");
+                            } else {
+                                showNotification(notificationObj);
+                            }
+                            // Break, we have acted on one notification already
+                            return false;
+                        }
```

Comment:
Probably we will need a `return true` here since without it, it will return `undefined` thereby making the `every` to stop iteration. I think `notifications.find` will be more suited to this usecase.

Reply:
Nice catch. This is important.

Reply:
Handled.